java/memory/src/main/java/org/apache/arrow/memory/ArrowByteBufAllocator.java

```diff
@@ -146,4 +147,39 @@ private RuntimeException fail() {
     throw new UnsupportedOperationException("Allocator doesn't support heap-based memory.");
   }
 
+  @Override
+  public int calculateNewCapacity(int minNewCapacity, int maxCapacity) {
```

Comment:
This method is copied from `io.netty.buffer.AbstractByteBufAllocator.calculateNewCapacity`

Reply:
add javadoc listing heritage and why